### ISSUE TITLE
Demo video

### ISSUE THREAD

**Issue by Dylan-Riley** [theanticrumpet/CounterpointGenerator#21]
Start with the tech stack.

"This is a console app written in dotnet core.
We eventually decided on dotnoet thanks to wide support and the dependency injection framework"

Talk about architecture
How easy is it to switch rules? Add new rules? Update ruleapplier?
Extracting rules from the book. It's on the devs to actually turn the flowery written text into actual computer language.

What did you want from the code? How did you achieve it architecture-wise?

There's gonna be too many solutions. How do we want to handle that? Filter out options?

Talk about the asynchronous stuff. Talk about singletons and race conditions.

**Comment by Dylan-Riley** [comment]
Once we finish talking about the project in its current state, also spend time talking about future plans and expansion for the project.

**Comment by Dylan-Riley** [comment]
Start by introducing the problem in the first place. We need to explain what a "counterpoint" and "cantus firmus" are, we need to explain how the text lays out rules and how that can even be turned into computer speak.

**Comment by Dylan-Riley** [comment]
### Rough outline:

Haoran: Into the project, define the problem and key terms

Dylan: Introduce text, talk about the idea of rules in the text, lead into how that gets translated into the program

Haoran: Talk about generator process, talk about inputs and outputs

Both: Continuing project work, expansion ideas

End: FLStudio demo

**Comment by Dylan-Riley** [comment]
Here's a rough draft of what I'm thinking of saying for the presentation

[script draft.pdf](https://github.com/theanticrumpet/CounterpointGenerator/files/6948200/script.draft.pdf)

-----

Updated version: 
[script draft updated.pdf](https://github.com/theanticrumpet/CounterpointGenerator/files/6949714/script.draft.updated.pdf)

**Comment by Dylan-Riley** [comment]
Unedited video demo of the output, in case someone wanted to see that. No VA and including the syncing clicks in the beginning which will be cut.

https://user-images.githubusercontent.com/57050171/128581583-6c98b059-013a-482e-ba10-a0b0acff8a8b.mp4

**Comment by Dylan-Riley** [comment]
Posted in Zoom chat by Aditi:

> Haoran: 
Counterpoint vs harmony: technical distinction perhaps not the most useful in introduction.
Consider sticking to something along "multiple notes/voices", can start with an example of polyphony like Mass for Five Voices.
Introduction of Fux's book: good!

> Dylan:
Introduction: some overlap with Haoran's introduction of the book so I like the point you make about computers being good at math but coordinate to smoothen the transition.
Extraction: pretty good!
Rule files: First motivate why we designed rules as we did - easily replacable implementations of an interface.

 > Haoran:
Generator part can divide into steps with diagram. Right level of detail.

 > Dylan:
I -> we?
